### PR TITLE
Add shorten extension

### PR DESCRIPTION
**Pull request by eminvergil**
## Description

# Shorten

Uses netlify's redirect functionality to make a personal URL shortener.

## References

- [Netlify Redirects](https://www.netlify.com/docs/redirects/)
- [Original Implementation](https://github.com/kentcdodds/netlify-shortener)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/shorten/src/shorten.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+type TShortenerPath = {
+  netlifyshortener: string;
+};
+
+type TData = {
+  url: string;
+};
+
+export default function Command() {
+  const [url, setUrl] = useState<string>("https://www.raycast.com/");
```
[the remark this answers is not in the document]
Comment (eminvergil) [reply extensions/shorten/src/shorten.tsx:24]: This is not needed and i should also add validation for url. thanks for suggestion 👍

File: extensions/shorten/src/shorten.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+type TData = {
+  url: string;
+};
+
+export default function Command() {
+  const [url, setUrl] = useState<string>("");
+  const preferences = getPreferenceValues<TShortenerPath>();
+
+  const netlifyShortenerPath = path.join(preferences.netlifyshortener, "/node_modules/netlify-shortener/dist/index.js");
```
Comment (jasonbarry) [inline extensions/shorten/src/shorten.tsx:28]: [q] Curious why the user needs to clone the Netlify shortener repo locally and set it as a preference -- could you add https://www.npmjs.com/package/netlify-shortener as a dependency of this extension and call it that way?

File: extensions/shorten/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Shorten
+
+Uses netlify's redirect functionality to make a personal URL shortener.
```
Comment (jasonbarry) [inline extensions/shorten/README.md:3]: Could you mention what exactly happens when you run this command here in the readme? Something like the following? 

CleanShot 2023-04-07 at 11 27 04@2x

Comment (jasonbarry) [reply extensions/shorten/README.md:3]: I believe this command must be run in a local Netlify site, is that correct?

**Pull request closed without merging**